server/api/db/sqldb/db.py

```diff
@@ -781,8 +774,8 @@ def read_artifact(
         artifact = db_artifact.full_object
 
         # If connected to a tag add it to metadata
-        if artifact_tag_uid:
-            self._set_tag_in_artifact_struct(artifact, computed_tag)
+        if tag:
+            self._set_tag_in_artifact_struct(artifact, tag)
```

Comment:
Looks like there's a change in behavior here - we'll always add tag to the object, while in the previous behavior a tag was added to the object only if the user explicitly asked for it.